protostar/commands/test/test_results.py

```diff
@@ -28,15 +31,25 @@ class TimedTestResult:
 
 
 @dataclass(frozen=True)
-class PassedTestCaseResult(TestCaseResult, TimedTestResult):
+class TimedTestCaseResult(TestCaseResult, TimedTestResult):
+    pass
```

Comment:
I don't think we need this extra level of abstraction.

Reply:
It's used to provide typing for the TestingSummary, however now I think that adding TimedTestResult as a parent of TestCaseResult might be a better idea.

Reply:
@kasperski95 wdyt?

Reply:
It might conflict with the FuzzTestCaseResult. Fuzz test case results could expect `execution_time` to be passed as a statistic.

Reply:
I've checked, there are no conflicts with fuzzing related results.

Reply:
Currently, there are no conflicts. At the other hand, time statistic can be unrelated to the total execution time. Inheritance should be ok.

Reply:
Reverted back to inheritance